Sources/SwiftFormatPrettyPrint/TokenStreamCreator.swift

```diff
@@ -2274,14 +2281,46 @@ private final class TokenStreamCreator: SyntaxVisitor {
   /// the expression, keeping the entire expression together when possible, before breaking inside
   /// the expression. This is a hand-crafted list of expressions that generally look better when the
   /// break(s) before the expression fire before breaks inside of the expression.
-  private func maybeGroupAroundSubexpression(_ expr: ExprSyntax) {
+  private func maybeGroupAroundSubexpression(
+    _ expr: ExprSyntax, combiningOperator operatorExpr: ExprSyntax? = nil
+  ) {
     switch expr {
-    case is FunctionCallExprSyntax, is MemberAccessExprSyntax, is SubscriptExprSyntax:
+    case is MemberAccessExprSyntax, is SubscriptExprSyntax:
       before(expr.firstToken, tokens: .open)
       after(expr.lastToken, tokens: .close)
     default:
       break
     }
+
+    // Function calls in expressions are conditionally grouped based on whether they are preceded by
+    // an assigning operator. Assigning the result of a function call is allowed to break the
```

Comment:
Instead of using the term "rectangle rule" here (since we don't define it anywhere, and the only references I can find online are in a couple of Google's public Java formatter references), let's replace the sentence starting with "Assigning the result of..." with:

> When a function call expression is assigned to an lvalue, we omit the group around the function call so that the callee and open parenthesis can remain on the same line, if they fit. This is a frequent enough case that the outcome looks better with the exception in place.

At some point we may want to formalize our use of the rectangle rule throughout the pretty printer, but for now, let's avoid it as "jargon."

Reply:
Good point, I forgot about the fact that rectangle rule isn't necessarily widely known.